src/main/java/org/jabref/gui/fieldeditors/GroupEditor.java

```diff
@@ -0,0 +1,57 @@
+package org.jabref.gui.fieldeditors;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.scene.input.TransferMode;
+
+import org.jabref.gui.DragAndDropDataFormats;
+import org.jabref.gui.autocompleter.SuggestionProvider;
+import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.PreferencesService;
+
+public class GroupEditor extends SimpleEditor {
+
+    private Optional<BibEntry> bibEntry;
+
+    public GroupEditor(final Field field,
+                       final SuggestionProvider<?> suggestionProvider,
+                       final FieldCheckers fieldCheckers,
+                       final PreferencesService preferences,
+                       final boolean isMultiLine) {
+        super(field, suggestionProvider, fieldCheckers, preferences, isMultiLine);
+
+        this.setOnDragOver(event -> {
+            if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP)) {
+                event.acceptTransferModes(TransferMode.MOVE);
+            }
+            event.consume();
+        });
+
+        this.setOnDragDropped(event -> {
+            boolean success = false;
+            if (event.getDragboard().hasContent(DragAndDropDataFormats.GROUP)) {
+                List<String> draggedGroups = (List<String>) event.getDragboard().getContent(DragAndDropDataFormats.GROUP);
+                if (bibEntry.isPresent()) {
+                    String newGroup = bibEntry.map(entry -> entry.getField(StandardField.GROUPS)
+                                                                     .map(oldGroups -> oldGroups + (preferences.getBibEntryPreferences().getKeywordSeparator()) + (draggedGroups.get(0)))
+                                                                     .orElse(draggedGroups.get(0)))
```

Comment:
Add a check that `draggedGroups.get(0)` isn't an empty string. If you drag the "All Entries" group in JabRef it will keep appending separators.